packages/sync/src/PersistentStore.ts

```diff
@@ -3,9 +3,9 @@
  * Interface for underlying storage solutions
  */
 export interface PersistentStore<T> {
-  getItem: (key: string) => Promise<T> | T;
-  setItem: (key: string, data: T) => Promise<void> | void;
-  removeItem: (key: string) => Promise<void> | void;
+  getItem: (key: string) => Promise<T>;
+  setItem: (key: string, data: T) => any;
+  removeItem: (key: string) => Promise<void>;
```

Comment:
It doesn't look for me that these changes were essential to get this work finished. Is that just an experiment or this is required?

Reply:
They are unfortunately. Our old interface didn't match Localforage exactly. There is a github issue somewhere, I'll try to find it

Reply:
It does not need to match local forage it needs to match the Apollo one. I will check what was failing in particular here and propose something. This one will break other providers as we made it less flexible rath